### PR TITLE
Add a separate install tools command

### PR DESCRIPTION
**Pull request by jsturtevant**
**Description**
Add a separate install tools command. This allows for something like `sudo az capi install` which fixes issues when installing things on Linux.  This seems to be the pattern according to https://github.com/Azure/azure-cli/issues/2558#issuecomment-289919156

Fixes: #90 

**History Notes**

Something I tried was using a temp path in the function `download_binary` if on linux but because this command was in a spinner it never prompted (it did but it got overwritten). It worked if I had recently used sudo.

```
  if system == "Linux":
         tmpfile = Path(tempfile.gettempdir()).joinpath(os.path.basename(install_location))
        urlretrieve(file_url, tmpfile)
        os.system(f"sudo mv {tmpfile} {install_location}")
  else:
```

---

This checklist is used to make sure that common guidelines for an Azure CLI pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by mboersma — COMMENTED** [review]
Makes sense to me, and we always want to follow `az` CLI conventions in general.

The linter is complaining about not having two blank lines between functions or classes. You can do `azdev style` to run it.

Lol, the comment you linked to was our fearless leader supporting `az acs kubernetes`. 😆 I want to call that "the good old days." but things have improved immensely since then.

**Review by DannyBrito — COMMENTED** [review]
This makes sense! I just have a question @mboersma. Do you think if we are having a separate method for installs, do we need to add `kind` into the list?

**Comment by jsturtevant** [comment]
@DannyBrito I was wondering something along the same lines.  Maybe this should implement it with an argument `--all` with the idea that later we could install individual components via an argument `--name` it was needed?

**Comment by mboersma** [comment]
I guess `kind` isn't always required, so we wouldn't want to install it by default. I like @jsturtevant's idea about making it installable with `az capi install --all` for now.

File: src/capi/azext_capi/_help.py [hunk 1]
```diff
@@ -141,3 +141,11 @@
 type: command
 short-summary: Update a CAPI management cluster.
 """
+
+helps['capi install'] = """
+type: command
+short-summary: installs all needed tools
+parameters:
+  - name: --all_tools
```
Comment (DannyBrito) [inline src/capi/azext_capi/_help.py:149]: If we are specifying the options_list in `_params.py` for `capi install` as `--all -a` need to refer in same way here to reflect changes in CLI  
```suggestion
  - name: --all -a
```

**Review by DannyBrito — APPROVED** [review]
lgtm

**Pull request merged**